lesson_3/src/com/freeit/methodicaltasks/Task21BuilderBufferString.java

```diff
@@ -0,0 +1,44 @@
+package com.freeit.methodicaltasks;
+
+/*
+Напишите три цикла выполняющих
+многократное сложение строк,
+один с помощью оператора сложения и String,
+другой с помощью StringBuilder
+и метода append,
+а также аналогино для StringBuffer.
+Сравните скорость их выполнения.
+*/
+public class Task21BuilderBufferString {
+    public static void main(String[] args) {
+
+        String string = new String();
```

Comment:
I detect that this code is problematic. According to the [Performance (PERFORMANCE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#performance-performance), [Dm: Method invokes inefficient new String() constructor (DM_STRING_VOID_CTOR)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#dm-method-invokes-inefficient-new-string-constructor-dm-string-void-ctor).
Creating a new java.lang.String object using the no-argument constructor wastes memory because the object so created will be functionally indistinguishable from the empty string constant "".  Java guarantees that identical string constants will be represented by the same String object.  Therefore, you should just use the empty string constant directly.

Reply:
thank you, i read the chapter deeper for a better understanding and try to correct it

Reply:
Happy to help.